src/Elastic.Apm/OpenTelemetry/ElasticActivityListener.cs

```diff
@@ -328,7 +337,87 @@ static string ToResourceName(string type, string name)
 			}
 		}
 
-		public void Dispose() => Listener?.Dispose();
+		private static bool TryGetStringValue(Activity activity, string key, out string value)
```

Comment:
Yeah, on reflection, I agree. Initially, I thought there only needed a single overload for a fallback, but since we can have 3+ key names to attempt, a static array for the set of names makes sense and might even clean up the call sites a bit. I will refactor that.

Reply:
I kept one overload so we can either pass a single key or many. This seemed useful to avoid defining single key arrays.